pkgs/by-name/ha/halo/package.nix

```diff
@@ -0,0 +1,61 @@
+{ lib
+, stdenv
+, fetchurl
+, makeWrapper
+, temurin-jre-bin
+, rsync
+}:
+stdenv.mkDerivation rec {
+  pname = "halo";
+  version = "2.15.1";
+  src = fetchurl {
+    url = "https://github.com/halo-dev/halo/releases/download/v${version}/${pname}-${version}.jar";
+    hash = "sha256-6qgbUXd0E38Rhv0Jf1qP54yJhQiB1a/Jr1FYw3bC4qI=";
+  };
+
+  nativeBuildInputs = [
+    makeWrapper
+    rsync
+    temurin-jre-bin
+  ];
+
+  dontPatch = true;
+  dontBuild = true;
+  dontConfigure = true;
+
+  unpackPhase = ''
+    cp $src ${pname}.jar
+    java -Djarmode=tools -jar ${pname}.jar extract --layers --launcher
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/application
+    rsync -a ${pname}/dependencies/ $out/application
+    rsync -a ${pname}/spring-boot-loader/ $out/application
+    rsync -a ${pname}/snapshot-dependencies/ $out/application
+    rsync -a ${pname}/application/ $out/application
+
+    mkdir -p $out/bin
+    makeWrapper ${temurin-jre-bin}/bin/java $out/bin/${pname} \
+      --chdir $out/application \
+      --set-default HALO_WORK_DIR "/var/lib/${pname}" \
+      --set-default JVM_OPTS "-Xms256m -Xmx256m" \
+      --set SPRING_CONFIG_LOCATION "optional:classpath:/;optional:file:\`\$HALO_WORK_DIR\`/" \
+      --add-flags "-server \$JVM_OPTS" \
+      --add-flags "org.springframework.boot.loader.launch.JarLauncher"
```

Comment:
Not sure if these options should be hardcoded here. It's better to put them in a module.